### PR TITLE
Add debian:stretch based Dockerfiles for 2.0

### PR DESCRIPTION
**Pull request by MichaelSimons**
1. Added new Debian Stretch based Dockerfiles for 2.0
2. Updated CI scripts to read manifest in order to determine what to build/test.  I would eventually like to remove the build script and replace it with the image-builder tool but that is currently blocked on some Windows/Docker issues.

### CODE REVIEW

File: build-and-test.ps1 [hunk 1]
```diff
@@ -6,48 +6,45 @@ param(
...
-        docker build $optionalDockerBuildArgs -t $tag $_.DirectoryName
-        if (-NOT $?) {
-            throw "Failed building $tag"
-        }
+$builtTags = [System.Collections.ArrayList]@()
+
+$manifestRepo.Images |
+    ForEach-Object {
+        $images = $_
+        ForEach-Object {$_.Platforms} |
```
Comment (dagood) [inline build-and-test.ps1]: This usage is very strange to me. Why not just `$_.Platforms |`? As far as I can tell, in this situation (`ForEach-Object` not given any inputs) `$_` keeps its value from the outside scope.

Comment (MichaelSimons) [inline build-and-test.ps1]: Your right this was a leftover from refactoring. Removed `ForEach-Object`

File: build-and-test.ps1 [hunk 2]
```diff
@@ -6,48 +6,45 @@ param(
...
+            ForEach-Object {
+                $dockerfilePath = $_.$platform.dockerfile
+                $tags = $_.$platform.Tags
+                if ([bool]($images.PSobject.Properties.name -match "sharedTags")) {
+                    $tags += $images.sharedTags
+                }
+
+                $qualifiedTags = $tags | ForEach-Object {
+                    $_ = $manifestRepo.Name + ':' + $_.Replace('$(nanoServerVersion)', $manifest.TagVariables.NanoServerVersion)
+                    $_
```
Comment (MichaelSimons) [inline build-and-test.ps1]: No reason - fixed.

File: build-and-test.ps1 [hunk 3]
```diff
@@ -6,48 +6,45 @@ param(
...
-                Replace("$PSScriptRoot$dirSeparator", '').
-                Replace("$dirSeparator$imageOs", '').
-                Replace($dirSeparator, '-')
-        $tags.Add($tag) | Out-Null
-        Write-Host "--- Building $tag from $($_.DirectoryName) ---"
-        docker build $optionalDockerBuildArgs -t $tag $_.DirectoryName
-        if (-NOT $?) {
-            throw "Failed building $tag"
-        }
+$builtTags = [System.Collections.ArrayList]@()
```
Comment (MichaelSimons) [inline build-and-test.ps1]: No reason - changed to array.

**Pull request merged**